app/payment-links/reference/reference.controller.js

```diff
@@ -50,18 +50,18 @@ function getPage (req, res, next) {
     return next(new NotFoundError('Attempted to access reference page with a product that auto-generates references.'))
   }
 
-  const sessionReferenceNumber = paymentLinkSession.getReference(req, product.externalId)
-
   if (paymentLinkSession.getError(req, product.externalId)) {
     const errors = {}
     const errorMessage = res.locals.__p(paymentLinkSession.getError(req, product.externalId))
     errors[PAYMENT_REFERENCE] = capitaliseFirstLetter(errorMessage)
     paymentLinkSession.setError(req, product.externalId, '')
     data.errors = errors
+    paymentLinkSession.removeReference(req, product.externalId)
```

Comment:
There is a unit test for this controller for the case where there is an error in the session. It would be good to add an assertion to this test that this method is called to clear the reference from the session.